server/datastore/mysql/app_configs.go

```diff
@@ -179,3 +179,15 @@ func getEnrollSecretsDB(ctx context.Context, q sqlx.QueryerContext, teamID *uint
 	}
 	return secrets, nil
 }
+
+func (ds *Datastore) AggregateEnrollSecretPerTeam(ctx context.Context) ([]*fleet.EnrollSecret, error) {
+	query := `
+          SELECT COALESCE(ANY_VALUE(secret), '') as secret, t.id as team_id FROM teams t LEFT JOIN enroll_secrets es ON t.id = es.team_id GROUP BY t.id
+        UNION
+          (SELECT COALESCE(secret,''), team_id FROM enroll_secrets WHERE team_id IS NULL LIMIT 1)`
```

Comment:
I wonder if it would be worth doing a sub-query instead and avoid the GROUP BY, making the result deterministic by always selecting the most recent (~ less likely to be removed, heuristically speaking) secret? E.g. something like that (totally untested, just to bounce ideas):

```
SELECT COALESCE(
SELECT es.secret FROM enroll_secrets es WHERE es.team_id = t.id ORDER BY es.created_at DESC LIMIT 1
, '') as secret, t.id as team_id
FROM teams t
```

and similar for no team?

Reply:
that's a great idea! I made the changes in 074fe71911de34f754f2633d31e85f176b188cb0 lmk if I got it right